### ISSUE TITLE
Notifications

### ISSUE THREAD

**Issue by onedaylove** [mfkp/tinderplusplus#21]
If you're able to, please add the ability to get notications when someone sends a message or when they have a match.

**Comment by mfkp** [comment]
Do you mean something like desktop notifications that pop up when there's a new message or match (even if the app is running in the background)?

**Comment by onedaylove** [comment]
I mean like the notifications you get on OS X with most apps. Notification sound, badge, banner, and possibly something in Notification Center. Not sure how that would work with Windows or Linux. I only use OS X.

**Comment by mfkp** [comment]
Yes - this appears like it would be possible: https://github.com/nwjs/nw.js/wiki/Notification

**Comment by mrmayfield** [comment]
+1, I'm going to dig into the source a bit today and see if I can get this up and running

**Comment by mrmayfield** [comment]
Added notifications for new matches

Still need to hook into notifications for new messages

Actually looking at it now, it loops through a forEach of each of the matches which probably isn't an ideal. Probably want to change the implementation or use a counter to only send one notification.

And basically, when update() is called in tinder++controls.js, there needs to be a check for new messages from getUpdate and then if they're valid you can safely send a new notification. Can't do that at the moment because out of 

```
{\"matches\":[],\"blocks\":[],\"matchmaker\":[],\"lists\":[],\"deleted_lists\":[],\"liked_messages\":[],\"last_activity_date\":\"2016-03-07T02:51:04.102Z\"}
```

I'm not sure which one represents new messages. Likely 'liked_messages' but I can't tell until I get a message and test this tomorrow.

**Comment by mrmayfield** [comment]
Nevermind. Just looked into it a bit more and the proper thing will be to check if the data.match.message is truthy, then fire off a notification saying you have a new message.

To make this nicer we could even perhaps add the name of the person ('You have a new message from Jessica!') and then on click would open that person's conversation. Not sure how to do that with nw.js , will have a chance to work on this again next weekend.
